### DISCUSSION TITLE
Maximum parallel async events or requests

### DISCUSSION THREAD

**Discussion by kaustubh-d** [aio-libs/aiohttp#5874] [Q&A]
Hi, Is there a limit on Maximum async events or requests that can be raised or handled in aiohttp based application? There are few suggestions online to use semaphore for controlling number of simultaneous requests, but what would be recommended count, what is maximum number for simultaneous async events?

**Comment by Dreamsorcerer** [comment]
I don't know the exact details, but "async events" is probably limited only by the amount of RAM you have (I'm not sure there's any reason for asyncio to try and restrict this). Number of connections in aiohttp, according to the documentation, defaults to 100:
https://docs.aiohttp.org/en/stable/client_advanced.html#limiting-connection-pool-size

So, once 100 connections are open and actively being used, other requests will wait until a connection is freed and added back into the pool (I imagine this is implemented as a queue, rather than semaphores).

**Reply by kaustubh-d** [reply]
Thanks for info on queuing the request w.r.t connection pool size. Yeah RAM size will limit, but was wondering if there are any other limits. In some tests I saw some events (get requests) are dropped or not received and app continued to run fine, no out-of-memory errors. I tried increasing timeout to find if some requests were dropped due to timeouts, `timeout = aiohttp.ClientTimeout(total=900)`  but it does not seem to be the case. Yet to check RAM usage, will see if I find anything. 

If I throttle the requests to max 1k simultaneous requests, no requests are dropped.

**Reply by Dreamsorcerer** [reply]
Well, I'd say that requests dropping is unexpected behaviour. If you can write a simple script to reproduce the problem, then please file a bug with that code.
